build.ps1

```diff
@@ -255,8 +255,23 @@ function Invoke-Program()
 
     if ($LastExitCode -ne 0)
     {
-      $callstack = @(Get-PSCallStack) -Join "`n"
-      throw "Command execution returned $LastExitCode. Call stack:`n$callstack"
+      $ErrorMessage = "Error: $([IO.Path]::GetFileName($Executable)) exited with code $($LastExitCode).`n"
+
+      $ErrorMessage += "Invocation:`n"
+      $ErrorMessage += "  $Executable $Args`n"
+
+      $ErrorMessage += "Call stack:`n"
+      foreach ($Frame in @(Get-PSCallStack)) {
+        $ErrorMessage += "  $Frame`n"
+      }
+
+      $ErrorMessage += "Environment:`n"
+      foreach ($EnvVar in ((Get-ChildItem env:*).GetEnumerator() | Sort-Object Key))
+      {
+        $ErrorMessage += "  $($EnvVar.Key)=$($EnvVar.Value)`n"
```

Comment:
Is this a bad idea because it may surface github tokens or other credentials in logs?

Reply:
Yeah, I think that it might expose tokens.  I'm not sure that this is acceptable for CI purposes.